libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -210,12 +213,35 @@ internal class MapboxTripSession(
                 ifNonNull(tripStatus.route?.routeOptions()?.coordinates()?.size) {
                     it - tripStatus.navigationStatus.nextWaypointIndex
                 } ?: 0
+
+            var triggerObserver = false
+            route?.let {
+                ifNonNull(it.legs()) { legs ->
+                    val currentLeg = legs[tripStatus.navigationStatus.legIndex]
+                    ifNonNull(currentLeg?.steps()) { steps ->
+                        val currentStep = steps[tripStatus.navigationStatus.stepIndex]
+                        var bannerInstructions =
+                            tripStatus.navigationStatus.bannerInstruction?.mapToDirectionsApi(
+                                currentStep
+                            )
+                        val state = tripStatus.navigationStatus.routeState.convertState()
```

Comment:
We need to check if the state isn't invalid before calling `convertState` and abort if it is. Refs https://github.com/mapbox/mapbox-navigation-android/pull/4482#issuecomment-870640460.